pkg/rollout/rollout.go

```diff
@@ -238,12 +280,37 @@ func (r *Rollout) updateAnnotations(svc *run.Service, stable, candidate string)
 		return svc
 	}
 
+	if r.requireRollback {
+		delete(svc.Metadata.Annotations, CandidateRevisionAnnotation)
+		svc.Metadata.Annotations[LastFailedCandidateRevisionAnnotation] = candidate
+	} else {
+		svc.Metadata.Annotations[CandidateRevisionAnnotation] = candidate
```

Comment:
move this outside if

Reply:
Since we no longer delete the candidate revision annotation, I have moved that line outside as suggested.